packages/terra-table/src/subcomponents/Section.jsx

```diff
@@ -122,42 +123,50 @@ function Section(props) {
 
   const theme = useContext(ThemeContext);
 
-  const handleMouseDown = (event) => {
+  const gridContext = useContext(GridContext);
+  const isGridContext = gridContext.role === GridConstants.GRID;
+
+  const hasSectionButton = isCollapsible && onSectionSelect;
+
+  const handleClick = () => {
     onSectionSelect(id);
-    event.stopPropagation();
   };
 
   return (
     <>
-      {/* Render section rows */}
-      {!isHidden && (
-      <tr
-        aria-rowindex={sectionRowIndex}
-        className={cx('header')}
-      >
-        <th
-          id={`${tableId}-${id}`}
-          align="left"
-          colSpan={displayedColumns.length}
-          role="columnheader"
-          scope="col"
+      <tbody>
+        {/* Render section rows */}
+        {!isHidden && (
+        <tr
+          aria-rowindex={sectionRowIndex}
+          className={cx('header')}
+          data-section-id={id}
         >
-          <SectionHeader
-            text={text}
-            isOpen={!isCollapsed}
-            isTitleFixed
-            onClick={isCollapsible && onSectionSelect ? handleMouseDown : undefined}
-          />
-        </th>
-      </tr>
-      )}
+          <th
+            id={`${tableId}-${id}`}
+            align="left"
+            colSpan={displayedColumns.length}
+            role="columnheader"
+            scope="col"
+            tabIndex={isGridContext && !hasSectionButton ? -1 : undefined}
+          >
+            <SectionHeader
+              text={text}
+              isOpen={hasSectionButton ? !isCollapsed : undefined}
+              isTitleFixed
+              onClick={hasSectionButton ? handleClick : undefined}
```

Comment:
I observed that if SPACE button was pressed on focused section, it does both - first opens the section, then closes it. Per discussion with @cm9361 it become clear that the second toggle is happening due to section header button being triggered as well.

I guess it can be fixed with onKeyDown event handler added to intercept a SPACE key press only: 
```suggestion
              onKeyDown={handleKeyDown}
```
where 
```
  const handleKeyDown = (event) => {
    const key = event.keyCode;

    // Intercept the SPACE key only
    if (key === KeyCode.KEY_SPACE) {
      onSectionSelect(id);
      event.preventDefault();
    }
  };
  ```

Reply:
I will have to make an update in terra-core to fix this behavior.  I don't think the update should be part of this PR.